### ISSUE TITLE
GUN is not a great brand name

### ISSUE THREAD

**Issue by merarischroeder** [amark/gun#716]
It's quite hard to search the internet for references to Gun. Sometimes I use "Gun DB", but still, due to the strong polarising debate of gun use in the USA, even "Gun DB" can turn up short. 

Perhaps there's an official search term we should use - perhaps "GunDB"?. If so, all the branding and all mentions of "GUN" should be changed to "GunDB". 

Otherwise, you might consider changing the brand name in the next 1-3 years. You might choose a word where it would be easy to rank #1 in internet searches. I think "Slack" did a good job in that regard.

Thanks.

**Comment by skoczen** [comment]
This is totally bike shed painting on my part, and I'm certain this conversation has been had ad-nausem, but I wanted to add on to this.  The name for Gun bothers me every single time I go to code in it.  Every time I talk about it.  Every time I mention it to a coworker.  

I get that it's an acronym, but it's an acronym for a word really loaded with violence that's **inappropriate to bring into most workplace settings**.

Saying something like "we're going to use gun for this" in an office with a non-technical coworker passing by results in WTF+fear, instead of "oh, that must be some tech thing."  Compare to "we're going to use postgreql for this."

When the library name is inadvertently causing reasonable people to worry for their safety in a normal conversation, it might be worth re-examining.

**Comment by amark** [comment]
Most people use gunDB as the hash/tag/seo then reference GUN (all caps, proper branding) in the text.

A database is a loaded gun, and a company's employees should treat users' data (and the programming of it) with the same care they would a real gun.

Most people who use GUN are Open Source. The ones who deploy, are taking real battles up that corporations and governments might try to crush (the Internet Archive, notabug.io, etc.).

If you are uncomfortable speaking in your work place, maybe that means you should quit. There are better battles to fight for, debating how fearful cubicle workers are is... not.

**Comment by mitra42** [comment]
@amark - please don't use us in this context, most people I speak to don't like the name, including here at the Internet Archive. I accept that you've chosen it, and that its too late to change, but having a name and branding, which (in my experience) causes a visceral negative reaction almost every time people hear it for the first time, is certainly not helping adoption.

**Comment by amark** [comment]
@mitra42 edited, per your request.

**Comment by skoczen** [comment]
@amark  Wow.  Yikes.  I'll see myself out.

**Comment by amark** [comment]
@skoczen could you clarify your comment that you think I'm angry? Was it my response here (which disagreed with you, but I'm not angry) or some other interaction? I've been gone on a vacation for 2 weeks and want to make sure there wasn't some other incident that happened in the community that I was not aware aware of.

**Comment by bugs181** [comment]
@all Do keep in mind, that without rebranding, there are upcoming projects called Joy (JoyDB) and a new idea I've been throwing around called GEARS.

**Comment by amark** [comment]
@bugs181 did anything unusual happen while I was gone, in terms of community rapport?

**Comment by bugs181** [comment]
@amark not to my knowledge. In fact, community has been fairly quiet. Just technical questions and answers.

**Comment by merarischroeder** [comment]
@amark 

I have only just started looking at GUN, and I'm starting here with zero credibility. That can grow over time. I hope to succeed with an objective debate. I have spent almost an hour in preparing this reply, fully intending to help.

Brands matter, but they can be changed. 

Your reaction to this issue seems to be quite emotive. I'm sure you're very attached to the brand, and for good reason, **you have done an amazing job**. In this topic, no one is judging you, and I hope you can be open-minded to see if your opinion can shift willfully.

**Comparing "Gun" to NULL-BRAND is a non-contest**. There is a click-bait mantra in the lean startup workd - "Fail fast". That sounds like being defeated (hence click-bait), but it's a name for an important principle. You can read about it online, but for GUN, it's best if you at least brainstorm some other ideas. Once you do that, you can fall in love with another brand, and that helps you, over time, let go of the old one. Then you can **compare "GUN" with good options, which is fair.**

A very easy way to link laterally for brand names, is to flick through [BrandCrowd](https://www.brandcrowd.com/logo-design). There are many other ways of course.

Once you have good short-list brand candidates, startup people like to test everything. Experiment, measure, quantify, and prove assumptions.

I propose, that in the coming months you test a few brands. This should be done in a few ways, here are two:

1) A survey of your closest contributors. Perhaps here, SurveyMonkey, or another platform you already use with this brand that has surveys built in.

2) Digital marketing testing - using GUN as your organisation name, but have split testing with landing pages that adjust your Product name to the 4 options. Target decision makers will be able to bring more revenue to your project. LinkedIn might be the best platform to test against Software Architects. This would require a marketing budget of ~$5k plus time to set up. 

I'm happy to help, and happy to chat further on Skype. I have brainstormed some options, which I can share with you privately.

**Comment by amark** [comment]
@merarischroeder wonderful! We have just the thing that we need help with:

 - GUN is a unopinionated library, but we're trying to build...
 - JOY an opinionated framework.

there was a long gitter about it, but the moral of the story is we think 90% to 95% of app developers will just use the framework (like with React, etc.), and GUN will fade into the background for the 10 or 5% of developers who prefer libraries.

We're not very good framework people, but you seem to to have an eye for all the right tricks and could use somebody like you to scaffold out that perfect contagiousness that all the React/other devs will be hyped up about :)

Good (and bad) news is JOY was only announced a month ago and is basically an empty/stubbed project repo under ERA user handle. So it needs a lot of filling out, but it also means first people in who take initiative on proposing API & stuff will get to dictate/lead the project.

https://github.com/eraeco/joydb

I know @ThinkingJoules has some dibs on it too, maybe the two of you should discuss.

**Comment by merarischroeder** [comment]
@amark 

I'll be happy to help with coding at some stage. I don't have too much volunteer time to spare in the coming months. I have some volunteer time for software architecture and marketing. Both take fewer total hours compared to the coding.

I can see JoyDb as your new brand for positioning this toward enterprise. Opinionated is good for dev speed. 

1. I take it you'll start with React, and move to Angular and VueJS in the future?
2. I wonder what the roadmap is for JoyDb. I have very specific things that I would be looking for. As I have mentioned in another issue post, replication from GUN to SQL-Storage is one.
3. I still believe I have some unique brand insights and ideas for your consideration. If Joy is only just starting now, this is the time to review, in case you discover something important. Can I catch you somewhere less public to send you the details? Maybe just an email? Mine is todd.hubers@gmail.com

My branding (not just the brand name) ideas help both to attract enterprise customers, but also coders.

Thanks

**Comment by bugs181** [comment]
> 2\. I have very specific things that I would be looking for

Could you perhaps point me to that post. I am eager to hear feedback about database features, as this is something I believe could be implemented in a modular fashion. I've kind of been patiently waiting for others to jump on the JoyDB front, although it appears nobody has taken up the baton as much as I'd like. Earlier, I announced a proposal and in doing so, took up the challenge.

I would like to collaborate with you further, if teaming up for marketing JoyDB (and alternative name proposals) is something you're interested in.

**Comment by amark** [comment]
@merarischroeder mark@gun.eco :) :)

**Comment by merarischroeder** [comment]
@bugs181 The database feature request: [https://github.com/amark/gun/issues/715](https://github.com/amark/gun/issues/715).

I'll send an email soon

**Comment by ThinkingJoules** [comment]
Sorry I missed this convo, I was out on vacation for a couple days. @bugs181  I see you added an issue to JOY, I would like to do a write up on some stuff (I have been trying to get to that for a few weeks now..). I am in the middle of trying to finalize the backbone of my GBase project which I am hoping we can use (at least conceptually) as a prototype for JOY. Eventually I want to be able to rebuild my structure using the JOY framework (and thus be able to deprecate my GBase project). I will try to add some preliminary thoughts to the JOY repo later today or tomorrow.

@merarischroeder If you have any other thoughts (besides #715) please feel free to email me or simply add some issues to the JoyDB repo. My email is julian@thevry.co

**Comment by merarischroeder** [comment]
Thanks @ThinkingJoules. I have forwarded you an email too.

**Comment by go1dfish** [comment]
I just wanted to chime in and say I love the name, but agree that it can be difficult to search for and second the recommendation to use gundb for searching and tagging.

Guns are not evil, and neither is gunDB.

They are tools whose purpose is defined by those who wield them.

> it's an acronym for a word really loaded with violence

Does your code use "state"?   https://www.hawaii.edu/powerkills/20TH.HTM

**Comment by mitra42** [comment]
But that's the problem @go1dfish - whatever your opinion or my opinion, at least 50%, (and in my experience a much higher percentage) of those who hear the word DO think Guns are evil (as compared to probably 5% who think the "state" is evil), and that means the brand (both name and logo) has a negative connotation in many, if not most, contexts. 

Every time I mention who we work with, I wait, for the negative reaction, which I don't believe is what you want from a brand.

**Comment by go1dfish** [comment]
An even higher percentage would say that **kill**ing is wrong, yet it's been a unix command for decades.

I've heavily embraced the name for my gun related libraries and I don't plan on changing them.

 * https://github.com/notabugio/gun-scope (more of a pun on the multiple meanings of scope)
 * https://github.com/notabugio/gun-receiver (conceptually similar to a firearm's receiver in design)
 * https://github.com/notabugio/gun-suppressor (validation works via suppressing messages, also a bit punny)
 * https://github.com/notabugio/gun-cleric (Because oracles are somewhat synonymous with clerics and I enjoyed the movie equilibrium)

**Comment by amark** [comment]
Thanks @go1dfish .

Maybe the people who care about naming so much for their non-Open Source corporate environment could pay a whitelabeling fee? I'm also sure the lawyers at those companies would be happy to avoid OSS for legal reasons too.

I'm happy to have ERA INC offer $50K / annual per-process license, with enterprise support, and whitelabeling (all `gun` naming will be scrubbed). This is not something I'm interested in having the company do (commercial license / enterprise support), but I think it is a fair solution.

Let me know. Else wise, I'd rather stick to the important things: providing real value, and true Open Source, to decentralize the web.

**Comment by amark** [comment]
<details><summary>This post contains sensitive material that may be offensive to some, do not click to view unless you believe in Free Speech.</summary>
<p>Anyone who is appalled at the thought of paying for code/whitelabel, and who thinks they are entitled to boss others who do the actual real work into their religious political views, are fascists masquerading around in their own moral fashion superiority.</p>
</details>

**Comment by mitra42** [comment]
@amark @go1dfish - I, for one, am not interested in getting into a political argument, the chance of either of us convincing the other is approximately zero and I respect your rights to your views. Nor am I suggesting you change the name you obviously love - again, as the code developers its your right to choose whatever name you wish. 

My question is how you deal with a situation where your brand (both the name and logo), you’ve chosen has strong negative connotations to half or more of the potential adopters. Look at how your initial response just drove away a potential volunteer, and I know this is far from the first time the issue has been raised.

**Comment by amark** [comment]
@mitra42 what do you mean? @merarischroeder has been very respectful and is already helping contribute, even though he knows I disagree with a variety of things.

**Comment by fineline** [comment]
Hi, I don't really care about the name but I see someone mention above that it is an acronym. I can't find a reference on the homepage or git readme or anywhere else for what this might be? Can anyone enlighten me? And are the other pieces - AXE, HAM etc. acronyms too? A handy [TLA](https://en.wikipedia.org/wiki/Three-letter_acronym) guide would be cool.

**Comment by bugs181** [comment]
@fineline I believe GUN stands for Graph Universal Node, HAM is Hypothetical Amnesia Machine, not quite sure what AXE means. I concur that a TLA guide would be handy.

**Comment by mitra42** [comment]
@amark I'm refering to @skoczen who raised the issue in the second comment and then said "Wow. Yikes. I'll see myself out." in response to your comment.

**Comment by amark** [comment]
@mitra42 given his angry attitude, it seems like he self-selected as not being a good fit. I'm not really worried about "losing a volunteer" that can't engage in civil dialogue - that is not what OSS or being a volunteer is about.

I've had pretty tense discussions with @go1dfish too, about things like his hatred of the versioning, but he and others have always at least engaged, cared, and been civil. The community is certainly not homogenous, but they at least actually contribute.

**Comment by amark** [comment]
@fineline welcome! :) First time I have heard from you, please feel free to jump on the chat room - I'd love to hear about what you are building, and more than happy to help you with anything.

:D all the names are backcronyms. @bugs181 pointed out a couple, here are the others:

 - Graph. Universe. Node.
 - Security, Encryption, Authorization.
 - RADix At Disk (also a recursive backronym!)
 - Daisy-chain Ad-hoc Mesh-network.
 - Advanced eXchange Equation
 - Economic Resource Allocation.

:D :P

**Comment by fineline** [comment]
Thanks all, I especially like Hypothetical Amnesia Machine - sounds like something from Hitchiker's Guide. Not building right now, just looking. Some interesting concepts here. I'll keep an eye on the project and see if it fits with anything that I'm doing in the future. All the best.

**Comment by merarischroeder** [comment]
**Backward**

@mitra42 is correct about varying opinions. My personal concern was not the content of the "Gun" debate, but the size of the debate, and the fact that it eclipses GunDB search results. There is definitely a problem here. Whether or not it is directly addressed is subject to @amark. 

@go1dfish You mentioned the `kill` unix command. Guns are actually more divisive and "triggering" than 'Killing'. (Also, sex is censored more than violence in anglo-western culture - I wonder if "SexyDb" would work!!)

@amark Whitelabelling is not a solution. The problem is the general brand, not specifically the enterprise application. Enterprise is only one context that has been provided, that highlights the issue. In addition to web-search, volunteers, and more. Even if you believe losing a volunteer here and there isn't a big problem, I believe it's a massive problem. To be relevant you need to be king of the hill, the most prominent and popular. To get there, you need every element of a marketing strategy to be streamlined toward growth.

**Forward**

JoyDB - or a new turnkey brand is certainly a step in the right direction. As I am finding out, GUN is just one of the many technical components of such a packaged solution.

Rather than a circling debate, I suggest we explore "JoyDb" further together, and the potential to create a new brand and channel to the public.

**Comment by bugs181** [comment]
@merarischroeder perhaps these are some of the things you'd like to discuss over at [Joy](https://github.com/eraeco/joydb). I don't think you'll get very far expressing your views on the "Gun" terminology here. As you've said, it's just one the many components in the technical stack that makes up GunDB. Kind of like the glue, if you will. GunDB holds them all together. 

Joy's approach is very similar and rather than working from the ground up, we will be working closely with the Gun team by making the database more modular. The front-facing API will be dramatically different but the core concepts remain the same. Realtime, decentralized, offline-first, mutable graph database. 

Querying the database is much different than what you see in Gun because of offering different Query languages that interface as store adapters. This means, that you can bring your traditional syntax/query language from your favorite database and ideally "it should just work". One additional Query syntax is available, and that's to use the modular parts themselves with the [Frame](https://github.com/frame-js/Frame) API.

In addition, Joy can be run as a service for server-side federated architecture while exposing its API via HTTP, WebSockets, etc. This will dramatically help with adoption.

We're all really excited to be in the planning stages of Joy and we're always on the lookout for new feature ideas. We encourage open discussion.

**Comment by merarischroeder** [comment]
@bugs181 I have shared background ideas in a Google Document. I used that method because of the branding discussion in there, which had a slight chance of being commercial in confidence.

Your description looks great. I'll try to contribute directly on that repo.

**Comment by bugs181** [comment]
Google docs are great for in-team collaboration but our objective with Joy is to be open and transparent. The community as a whole get's to set the guidelines. Anyone who disagrees with this philosophy will essentially be outcast and a fork will be made if it ever came down to it.

**Comment by merarischroeder** [comment]
They're just private thoughts to consider for the founding team and the founding time. If we discuss a brand, and we all say we like it, you might find some people decide to squat on cyber property and sell it to us at a higher price. 

I'm doing it to protect the content and potential brand. Once those conversations have been made, it can become public I guess. But only after the IP strategy and branding has been cemented.

I hope you can see that very few philosophies can be applied universally, there are usually exceptions to be discovered and integrated. I understand your intentions for your philosophy, but as you can see, temporary private conversations can be necessary and warranted. I hope you find a strategic short-term private discussion to be tolerable.

**Comment by miguelmota** [comment]
As a user of GunDB I have no issues with the name and love the fact that it's short and easy to say. Haven't experienced difficulties in searching docs either. Agree that it's difficult to market and has a negative connotation to some people and branding should maybe be changed to GunDB rather than just GUN but it's not the worst thing in the world as it is.

**Comment by resession** [comment]
"gunDB" -> works
change the name from "gunDB" to "shitDB"
"shitDB" -> works
change to "assholeDB"
"assholeDB" -> works
change to "fluffyDB"
"fluffyDB" -> works

HAHHAHA what the hell, its a package/database that makes apps more p2p and decentralized. if you need something like that you use this package......i can understand the uncomfortable feeling at the first few seconds, but after some seconds and some research the solutions/benefits that gundb might give you should override the bad points with the name. if it doesn't override the uncomfortable feeling, than hey more p2p and decentralized packages for me.....i got a fever, and the only prescription is more decentralized solutions.......if you do change the name though, i'll put in a suggestion. pamelaDB, cuz then you get thoughts of pamela anderson's old breasts. if name matters so much

**Comment by resession** [comment]
if you change the name than i have a suggestion. BigCockDB, whats better than having a big gun, thats right having a huge cock

**Comment by amark** [comment]
hey, person who posted ~1 hour ago: your language is crude, I do not like it, and that is **not** how our community operates. I would like to delete your comment, tho I disagree with censoring people, GitHub does grant me the ability to edit, so I will probably edit your post to make it difficult for people to read unless they want to opt-into crude language and/or fact-check me.
